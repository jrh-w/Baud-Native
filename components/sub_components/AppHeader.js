import React, { Component } from 'react';

import { Card, CardItem, Body, H1, Text, Left } from 'native-base';
import { Col, Row, Grid } from 'react-native-easy-grid';
import { Image, Dimensions } from 'react-native';

class AppHeader extends Component {
  constructor(props) {
    super(props);
  }

  render() {
    return(
        <Card style={{ marginTop: 0, paddingTop: 60 }}>
          <CardItem>
            <Body style={{ alignItems: 'center' }}>
              <Row>
                <Image style={{ width: Dimensions.get('window').height*0.2, height: Dimensions.get('window').height*0.2 }} source={require('../../assets/logo.png')} />
              </Row>
              <Row>
                <H1 style={{ marginTop: 20, marginBottom: 10}}>
                  Baud
                </H1>
              </Row>
            </Body>
          </CardItem>
        </Card>
    );
  }
}

export default AppHeader;
