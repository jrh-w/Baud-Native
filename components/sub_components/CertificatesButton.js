import React, { Component } from 'react';
import { View, Text } from 'react-native';

import { Card, CardItem, Body, H1, Icon, ProgressBar } from 'native-base';
import { Col, Row, Grid } from 'react-native-easy-grid';

class CertificatesButton extends Component {
  constructor(props) {
    super(props);
  }

  render() {

    return(
      /*<Card style={{ width: 150, marginRight: 20 }}>
        <CardItem style={{ backgroundColor: this.props.bgColor, textColor: this.props.color }}>
          <Body style={{ alignItems: 'center' }}>
            <Col>
              <Body>
                <Row>
                  <Icon type='Ionicons' name={this.props.icon} />
                  <H1 style={{ fontSize: 12, marginLeft: 10 }}>
                    {this.props.name}
                  </H1>
                </Row>
                  <Row style={{ marginTop: 5 }}>
                  <H1 style={{ fontSize: 15 }}>
                    {this.props.progress >= 100 ?
                      'COMPLETED'
                      :
                      <ProgressBar progress={this.props.progress} />
                    }
                  </H1>
                </Row>
              </Body>
            </Col>
          </Body>
        </CardItem>
      </Card>*/
      <View>
      <Card style={{ width: 150, marginHorizontal: 10 }}>
        <CardItem style={{ backgroundColor: this.props.bgColor, borderColor: this.props.bgColor }}>
          <Body style={{ alignItems: 'center' }}>
            <Col>
              <Body>
                <Row>
                  <Icon type='Ionicons' name={this.props.icon} style={{ color: this.props.color }}/>
                  <H1 style={{ fontSize: 12, marginLeft: 10, color: this.props.color }}>
                    {this.props.name}
                  </H1>
                </Row>
                <Row style={{ marginTop: 5 }}>
                  {this.props.progress >= 100 ?
                  <H1 style={{ fontSize: 13, color: this.props.color }}>
                    COMPLETED
                  </H1>
                  :
                  <H1 style={{ fontSize: 13, color: this.props.color }}>
                    UNCOMPLETED
                  </H1>
                  }
                </Row>
              </Body>
            </Col>
          </Body>
        </CardItem>
      </Card>
      </View>
    );
  }
}

export default CertificatesButton;
