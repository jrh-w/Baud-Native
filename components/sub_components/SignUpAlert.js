import React, { Component } from 'react';
import { View, Dimensions } from 'react-native';

import { Card, CardItem, Body, Text } from 'native-base';
import { Col, Row, Grid } from 'react-native-easy-grid';

class SignUpAlert extends Component {
  constructor(props) {
    super(props);
  }

  render() {

    return(
      <View>
      {this.props.errorText.length > 0 ?
      <Card style={{ backgroundColor: '#33333300', marginTop: 5, marginBottom: 5}}>
        <CardItem style={{ borderWidth: 1, borderColor: '#FF595E', height: Dimensions.get('window').height*0.08 }}>
          <Text style={{ color: '#FF595E', textAlign: 'center', fontSize: 14, marginHorizontal: 10, marginVertical: 5 }}>
            {this.props.errorText}
          </Text>
        </CardItem>
      </Card>
      :
      null
      }
      </View>
    );
  }
}

export default SignUpAlert;
