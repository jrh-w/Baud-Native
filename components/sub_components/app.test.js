import React from 'react';
import SignUpAlert from './SignUpAlert';
import CertificatesButton from './CertificatesButton';
import AppHeader from './AppHeader';
import renderer from 'react-test-renderer';

test('renders correctly', async () => {
  const tree = renderer.create(<SignUpAlert />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders correctly', async () => {
  const tree = renderer.create(<CertificatesButton />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders correctly', async () => {
  const tree = renderer.create(<AppHeader />).toJSON();
  expect(tree).toMatchSnapshot();
});
