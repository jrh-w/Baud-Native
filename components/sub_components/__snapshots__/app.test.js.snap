// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `<View />`;

exports[`renders correctly 2`] = `
<View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
          "borderBottomWidth": 0.5,
          "borderColor": "#ccc",
          "borderLeftWidth": 0.5,
          "borderRadius": 2,
          "borderRightWidth": 0.5,
          "borderTopWidth": 0.5,
          "elevation": 3,
          "flexWrap": "nowrap",
          "marginBottom": 5,
          "marginLeft": 2,
          "marginRight": 2,
          "marginTop": 5,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.1,
          "shadowRadius": 1.5,
        },
        Object {
          "marginHorizontal": 10,
          "width": 150,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "borderRadius": 2,
            "flexDirection": "row",
            "paddingBottom": 10,
            "paddingLeft": 15,
            "paddingRight": 15,
            "paddingTop": 10,
          },
          Object {
            "backgroundColor": undefined,
            "borderColor": undefined,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "flex-start",
              "alignSelf": "stretch",
              "flex": 1,
            },
            Object {
              "alignItems": "center",
            },
          ]
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "column",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "flex": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                  "flexDirection": "row",
                }
              }
            >
              <Text />
              <Text
                style={
                  Array [
                    Object {
                      "color": "#000",
                      "fontSize": 27,
                      "lineHeight": 32,
                    },
                    Object {
                      "color": undefined,
                      "fontSize": 12,
                      "marginLeft": 10,
                    },
                  ]
                }
              />
            </View>
            <View
              style={
                Object {
                  "flex": 1,
                  "flexDirection": "row",
                  "marginTop": 5,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#000",
                      "fontSize": 27,
                      "lineHeight": 32,
                    },
                    Object {
                      "color": undefined,
                      "fontSize": 13,
                    },
                  ]
                }
              >
                UNCOMPLETED
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`renders correctly 3`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "borderBottomWidth": 0.5,
        "borderColor": "#ccc",
        "borderLeftWidth": 0.5,
        "borderRadius": 2,
        "borderRightWidth": 0.5,
        "borderTopWidth": 0.5,
        "elevation": 3,
        "flexWrap": "nowrap",
        "marginBottom": 5,
        "marginLeft": 2,
        "marginRight": 2,
        "marginTop": 5,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.1,
        "shadowRadius": 1.5,
      },
      Object {
        "marginTop": -80,
        "paddingTop": 80,
      },
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderRadius": 2,
        "flexDirection": "row",
        "paddingBottom": 10,
        "paddingLeft": 15,
        "paddingRight": 15,
        "paddingTop": 10,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "flex-start",
            "alignSelf": "stretch",
            "flex": 1,
          },
          Object {
            "alignItems": "center",
          },
        ]
      }
    >
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "height": 266.8,
              "width": 266.8,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#000",
                "fontSize": 27,
                "lineHeight": 32,
              },
              Object {
                "marginBottom": 10,
                "marginTop": 20,
              },
            ]
          }
        >
          Baud
        </Text>
      </View>
    </View>
  </View>
</View>
`;
